jetty-server/src/main/java/org/eclipse/jetty/server/session/Session.java

```diff
@@ -813,19 +823,66 @@ public void renewId(HttpServletRequest request)
         String extendedId = null;
         try (Lock lock = _lock.lock())
         {
-            checkValidForWrite(); //don't renew id on a session that is not valid
+            while (true)
```

Comment:
This loops seems superfluous.  The switch statement will just spin if the _state isn't one of the cases declared.

Reply:
Added default clause to throw IllegalStateException